pkg/app/ops/apikeylastusedtimeupdater/apikeylastusedtimeupdater.go

```diff
@@ -0,0 +1,140 @@
+	"github.com/pipe-cd/pipecd/pkg/model"
+	"github.com/pipe-cd/pipecd/pkg/redis"
+)
+
+var (
+	interval = 10 * time.Minute
+)
+
+type apiKeyStore interface {
+	List(ctx context.Context, opts datastore.ListOptions) ([]*model.APIKey, error)
+	UpdateLastUsedAt(ctx context.Context, id string, time int64) error
+}
+
+type apiKeyLastUsedTimeCache interface {
+	Get(k string) (interface{}, error)
+}
+
+type APIKeyLastUsedTimeUpdater struct {
+	apiKeyStore             apiKeyStore
+	apiKeyLastUsedTimeCache apiKeyLastUsedTimeCache
+	logger                  *zap.Logger
+}
+
+const apiKeyLastUsedCacheHashKey = "HASHKEY:PIPED:API_KEYS" //nolint:gosec
+
+func NewAPIKeyLastUsedTimeUpdater(
+	ds datastore.DataStore,
+	rd redis.Redis,
+	logger *zap.Logger,
+) *APIKeyLastUsedTimeUpdater {
+	return &APIKeyLastUsedTimeUpdater{
+		apiKeyStore:             datastore.NewAPIKeyStore(ds, datastore.OpsCommander),
+		apiKeyLastUsedTimeCache: rediscache.NewHashCache(rd, apiKeyLastUsedCacheHashKey),
+		logger:                  logger.Named("api-key-last-used-time-updater"),
+	}
+}
+
+func (c *APIKeyLastUsedTimeUpdater) Run(ctx context.Context) error {
+	c.logger.Info("start running APIKeyLastUsedTimeUpdater")
+
+	t := time.NewTicker(interval)
+	for {
+		select {
+		case <-ctx.Done():
+			c.logger.Info("APIKeyLastUsedTimeUpdater has been stopped")
+			return nil
+
+		case <-t.C:
+			start := time.Now()
+			if err := c.updateAPIKeyLastUsedTime(ctx); err == nil {
+				c.logger.Info("successfully update api key last used time", zap.Duration("duration", time.Since(start)))
+			}
+		}
+	}
+}
+
+func (c *APIKeyLastUsedTimeUpdater) updateAPIKeyLastUsedTime(ctx context.Context) error {
+
+	opts := datastore.ListOptions{
+		Filters: []datastore.ListFilter{
+			{
+				Field:    "Disabled",
+				Operator: datastore.OperatorEqual,
+				Value:    false,
+			},
+		},
+	}
+
+	apiKeys, err := c.apiKeyStore.List(ctx, opts)
+	if err != nil {
+		c.logger.Error("failed to list API key", zap.Error(err))
+		return err
+	}
+
+	for _, apiKey := range apiKeys {
+		lastUsedTimeByte, err := c.apiKeyLastUsedTimeCache.Get(apiKey.Id)
+		if err != nil {
+			c.logger.Error("failed to update last used time",
+				zap.String("id", apiKey.Id),
+				zap.Error(err),
+			)
+			continue
+		}
+
+		lastUsedTime, err := bytes2int64(lastUsedTimeByte.([]byte))
+		if err != nil {
+			c.logger.Error("failed to update last used time",
+				zap.String("id", apiKey.Id),
+				zap.Error(err),
+			)
+			continue
+		}
+		if lastUsedTime == apiKey.LastUsedAt {
+			continue
+		}
+		if err := c.apiKeyStore.UpdateLastUsedAt(ctx, apiKey.Id, lastUsedTime); err != nil {
+			c.logger.Error("failed to update last used time",
+				zap.String("id", apiKey.Id),
+				zap.Error(err),
+			)
+			continue
+		}
+	}
+
+	return nil
+}
+
+func bytes2int64(bytes []byte) (int64, error) {
+	numString := string(bytes)
+	num, err := strconv.ParseInt(numString, 10, 64)
+	if err != nil {
+		return 0, err
+	}
+	return num, nil
```

Comment:
```suggestion
	return strconv.ParseInt(numString, 10, 64)
```
and besides, not sure why do we not just remove this function and do this inline at L110